### ISSUE TITLE
Touch Screen stopped working after Google Play Service updated

### ISSUE THREAD

**Issue by Stargazzer** [htruong/crankshaft#352]
If you need support: Please post on https://www.reddit.com/r/crankshaft/ as this is a bug tracker, not a support forum.

Note: Reporting issues not based on official published builds will be closed without any comment.

- What phone do you have?

- What OS version?

- What crankshaft version?

- Steps to reproduce the bug, if you're planning to report a bug. Please indicate whether the bug is always repoducible.

- Please provide any further information that you might find helpful if available.

- Starting with pre4: Boot into dev mode (can be selected in settings).
  Reproduce the possible bug. Create the debug logs by pushing the grey button on bottom of screen.
  Add the created file (/boot/debug.zip) to the issue.

  Warning: if you have configured wifi passwords they will be blanked by script during creation.
  We can't make absolutly sure you passwords are removed so do a personal check of the files inside the zip.

**Comment by Stargazzer** [comment]
I have a raspberry pi3 b+ and the official touchscreen 7" display which worked flawlessly until the point where touchscreen works only in the chrankshaft menu but not in Android Auto like it did until this point.

Everything worked fine until Google Play Services updated.

I have run into a Catch-22.

I removed the Google Play Service updates successfully.

Android Auto won't work unless Google Play Services is updated.

Updated Google Play Services makes my touch screen not work in Android Auto.

Phone: Umidigi Z2 Pro

Android Version: 8.1.0

Build Number: UMIDIGI_Z2_PRO_V1.020180823

Raspberry Pi 3+ with 7" Touch Screen

CrankShaft: Crankshaft OpenCarOS NG Alpha-5 2019-06-08

**Comment by Novocaine85** [comment]
Same identical issue on my Galaxy Note 8 EU version.

Android version: 9.0

Build number: PPR1.180610.011.hadesRom_OneUI_v7.0

Raspberry Pi 3+ with 7" Touch Screen

CrankShaft: Crankshaft OpenCarOS NG Alpha-5 2019-06-08

**Comment by JoelBrenstrum** [comment]
Same issue on my OnePlus 5T

Android version: 9.0

Raspberry Pi 3+ with generic 7" screen and digitiser ("USB_TouchController (eGalax TouchScreen)")

X11: True

CrankShaft: Crankshaft OpenCarOS NG Alpha-5 2019-06-08

EDIT: I've attached the debug.zip
[debug.zip](https://github.com/opencardev/crankshaft/files/3316355/debug.zip)

**Comment by Getuwer** [comment]
I got the same issue on my Asus X00QD

Android version: 9.0

Raspberry Pi 3+ with 7" Touch Screen SKU:Z-0017 

Crankshaft OpenCarOS NG Alpha-4 2019-03-03
Crankshaft OpenCarOS NG Alpha-5 2019-06-08

Crankshaft : both not working. I was using the Alpha-4 wich worked using touchscreen on Android Auto and crankshaft . 
One day Android Auto touchscreen was not responding but touchscreen on crankshaft worked and playing Music with Spotify on Android Auto by starting Spotify via Android (smartphone) worked too.

I updated to Alpha-5 and this has had no changes.

[debug.zip](https://github.com/opencardev/crankshaft/files/3317174/debug.zip)

**Comment by hawkeyexp** [comment]
needs some debug what was changed by play services....

**Comment by chucktastic** [comment]
the f1xpl/aasdk repository had an issue report a few days ago stating the sdk would no longer work with android auto due to a protocol buffer incompatibility.  Something about AA using proto2 and aasdk using proto3 and play services starting to be more strict about the data it receives.  Issue reports have been disabled on f1xpl/aasdk for some reason and the debug for that issue is no longer visible.

Hopefully that helps a little.

**Comment by Getuwer** [comment]
I tryed to do debug files 

I uninstalled Play Services updates on my phone and i catch some apk from apkmirror.com 

I first installed Google Play Services _17.4.55_ (100400-248795830) version, i plug my phone to my rpi and Android Auto started and touch screen was working , i then using touchscreen returning to crankshaft-ng menu and do debug file using the button debug while usb still plug

Then installed Google Play Services _17.5.28_ (100400-252519129) version, i plug my phone to my rpi and Android Auto started and touch screen was working , i then using touchscreen returning to crankshaft-ng menu and do debug file using the button debug while usb still plug 

Then installed Google Play Services _17.7.84_ (100400-252519124) version, i plug my phone to my rpi and Android Auto started but  **touch screen was not working anymore in AA** so i cannot reach crankshaft-ng menu with my touchscreen so i unplugged my phone , launched a debug and pluged my phone right after , AA started and debug was still in progress . 
So i got a debug file but it should not be really good. I should do my debug using ssh when my phone is plugged but i do not now how to debug via ssh.


[debug_working_17.4.55.zip](https://github.com/opencardev/crankshaft/files/3317911/debug_working_17.4.55.zip)
[debug_working_17.5.28.zip](https://github.com/opencardev/crankshaft/files/3317912/debug_working_17.5.28.zip)
[debug_not_working_17.7.84.zip](https://github.com/opencardev/crankshaft/files/3317910/debug_not_working_17.7.84.zip)

**Comment by ishan747** [comment]
Same identical issue on my Galaxy S8 Qualcomm 

Android version: 9.0

Raspberry Pi 3 B+ with Other 7" touchscreen

CrankShaft: Crankshaft OpenCarOS NG Alpha-5 2019-06-08

**Comment by iofirag** [comment]
Same identical issue on my Galaxy S7

Android version: 7.0

Raspberry Pi 3 B with  7" touchscreen

CrankShaft: 2019-03-03-crankshaft-ng-299C16C.zip
Google Play Services version 17.7.85

Touchscreen + mouse is working fine in the crankshaft menu's. but both stops working when connecting Phone.

**Comment by ulido** [comment]
Same issue with my Nokia 7 Plus

Android version: 9

RPi 3 A+ with official 7" touchscreen

CrankShaft latest
Android Auto version 4.3.591854
Google Play Services version 17.7.85

**Comment by xxxjajoxxx** [comment]
Same issue with my Xiaomi Mi6

Android version: 8.0

RPi 3 B+ with official 7" touchscreen

CrankShaft latest
Android Auto version 4.1.590944
Google Play Services version 17.7.85 (040408-253824076)

**Comment by eye2web** [comment]
Same issue with Huawei mate 20 pro
Android version: 9.0

RPi 3 B+ with 10.1" Waveshare touchscreen

CrankShaft latest
Android auto 4.4.592344
Google play services 17.7.85

Touchscreen is working in the crankshaft menu's. And stops working when connecting Phone.

**Comment by tompeel** [comment]
Same Issue
Pixel 2XL
Android Q beta Beta 4
GPS - 17.7.85 (100400-253824076)
AA - 4.3.591854

**Comment by xxxjajoxxx** [comment]
downgraded google play services to 17.1.22 and touchscreen went crazy random touches all over the android auto app works fine in crankshaft

**Comment by josemaes** [comment]
Same issue.
Samsung S8+ Android 9

CrankShaft: 2019-03-03-crankshaft-ng-299C16C.zip

7" screen + USB touch panel

**Comment by KK1423** [comment]
I am experiencing the same issue.

Phone: Oneplus 6T

Crankshaft: May 2018

**Comment by KK1423** [comment]
A workaround is to open the application settings for Google Play Services and uninstalling all updates. Maybe a newer apk could be sideloaded, but the factory version is all that I tested.

**Comment by Richard6360** [comment]
Same here on samsung A3 2017
Cant un install google play services

Op vr 28 jun. 2019 01:07 schreef Karthik Karyamapudi <
notifications@github.com>:

**Comment by gcharlie** [comment]
I'll pull a debug later tonight, but I just got the update on my Google pixel 3 and same is happening to me. 
Additionally I noticed that Android auto had also stopped responding to  gpio mapped button commands

**Comment by ashwaganda15** [comment]
Same.
Samsung S9+
Android 9
7" touchscreen
crankshaft 4 8 18

Work around is to use "OK Google" and tell AA to play the music you want.
 
now to reflash crankshaft as I figured my sd was corrupt.

Thanks for the info everyone-  I'll wait for someone smarter to post the fix...hopefully.

**Comment by SonofAdam** [comment]
Same here, Pixel XL  after google play services update touch screen stopped working .

**Comment by atjohnson23** [comment]
Same issue: 

Phone - Samsung S9 Plus on Android 9.0
CrankShaft -  latest
Android auto - 4.4.592344
Google play services - 17.7.85

Definitively a problem with google play services update it seems. I will downgrade and report results.

**Comment by tompeel** [comment]
Im having issues downgrading my play services, with out formatting the phone its basically not going to happen so cant test this temporary fix.
Any progress on a fix?

**Comment by josemaes** [comment]
> Im having issues downgrading my play services, with out formatting the phone its basically not going to happen so cant test this temporary fix.
> Any progress on a fix?

You have to disable "locate my phone" and google pay in the device administrator apps. Then you can remove the app updates but, even I disabled automatic updates and offline data for the app, it updates again after some hours. It is a very temporary fix.

**Comment by rhatfield-bushel** [comment]
I got this working for the time, so I figured I'd share.

First you'll need to download the google play services APK version for your phone, here's the guide I used (pay attention to the sub version numbers):
https://itechify.com/2018/01/12/install-right-google-play-services-android-device/
And the apk downloads link:
https://google-play-services.en.uptodown.com/android/versions

I tried `17.7.81` but that didn't work, I ended up using `17.5.26`.
Here's the settings I changed:
* `Settings > Google Services > Location` turn off location entirely
* `Settings > Google Services > Security > Find my device` turn off entirely
* `Settings > Apps > App Info > Google Play Services` first disable, then force stop
* `Settings > Apps > App Info > Google Play Services > Storage > Manage Space`  Clear all data, then on the previous screen Clear Cache

At this point your phone will likely be freaking out with tons of notifications. I'd disable all sounds before you start, it gets really annoying with it buzzing every 2 seconds.

Enable the app again, and install the APK you downloaded. If disable worked, it should install fine.

Go to Google Store (before or after install) and disable auto update. I think side loading an APK might prevent auto update anyways.

You'll need to re-enable all location services, but just open the Android Auto app and let it do it's thing. This works for me so far, we'll see if it tries to update itself anyways.

I got this working on the 7" touchscreen, Raspberry Pi 3 B, standard release SD image, and a Moto Z2 Force phone.

**Comment by CodeFoodPixels** [comment]
Just as a warning, rolling back google play services may break other apps or stop them from working correctly.

**Comment by goon818** [comment]
The touch don't work with my phone ans with crankshaft.
I testet die Viktorgino headunit Project. The Touch working with it. I hope this information can help to find the problem.

**Comment by gregorhollmig** [comment]
Is this issue openauto related? 
Is the openauto fork used in crankshaft? It seems the openauto project does not get any source updates anymore. maybe this could be helpful? 
https://github.com/f1xpl/openauto/pull/170

**Comment by krisstakos** [comment]
> 
> 
> I got this working for the time, so I figured I'd share.
> 
> First you'll need to download the google play services APK version for your phone, here's the guide I used (pay attention to the sub version numbers):
> https://itechify.com/2018/01/12/install-right-google-play-services-android-device/
> And the apk downloads link:
> https://google-play-services.en.uptodown.com/android/versions
> 
> I tried `17.7.81` but that didn't work, I ended up using `17.5.26`.
> Here's the settings I changed:
> 
>     * `Settings > Google Services > Location` turn off location entirely
> 
>     * `Settings > Google Services > Security > Find my device` turn off entirely
> 
>     * `Settings > Apps > App Info > Google Play Services` first disable, then force stop
> 
>     * `Settings > Apps > App Info > Google Play Services > Storage > Manage Space`  Clear all data, then on the previous screen Clear Cache
> 
> 
> At this point your phone will likely be freaking out with tons of notifications. I'd disable all sounds before you start, it gets really annoying with it buzzing every 2 seconds.
> 
> Enable the app again, and install the APK you downloaded. If disable worked, it should install fine.
> 
> Go to Google Store (before or after install) and disable auto update. I think side loading an APK might prevent auto update anyways.
> 
> You'll need to re-enable all location services, but just open the Android Auto app and let it do it's thing. This works for me so far, we'll see if it tries to update itself anyways.
> 
> I got this working on the 7" touchscreen, Raspberry Pi 3 B, standard release SD image, and a Moto Z2 Force phone.

I roll back the Google Services but after a while they updated again. Even after I turned background data usage off

**Comment by hawkeyexp** [comment]
> 
> 
> Is this issue openauto related?
> Is the openauto fork used in crankshaft? It seems the openauto project does not get any source updates anymore. maybe this could be helpful?
> [f1xpl/openauto#170](https://github.com/f1xpl/openauto/pull/170)

Yes the problem is related to openauto or aasdk interacting with play services. The ping problem is solved longer time ago - only the touch problem was a new effect with latest play service update. Openauto has stopped any source updates (the original announce was the gpl version will get fixes also when openauto pro - commercial - is available - but as you will see - no updates for gpl based version.

But i try the changes cause we skipped the ping pong to fix the bug which was working for long time now. Thx for the hint.

**Comment by xxxjajoxxx** [comment]
> > I got this working for the time, so I figured I'd share.
> > First you'll need to download the google play services APK version for your phone, here's the guide I used (pay attention to the sub version numbers):
> > https://itechify.com/2018/01/12/install-right-google-play-services-android-device/
> > And the apk downloads link:
> > https://google-play-services.en.uptodown.com/android/versions
> > I tried `17.7.81` but that didn't work, I ended up using `17.5.26`.
> > Here's the settings I changed:
> > ```
> > * `Settings > Google Services > Location` turn off location entirely
> > 
> > * `Settings > Google Services > Security > Find my device` turn off entirely
> > 
> > * `Settings > Apps > App Info > Google Play Services` first disable, then force stop
> > 
> > * `Settings > Apps > App Info > Google Play Services > Storage > Manage Space`  Clear all data, then on the previous screen Clear Cache
> > ```
> > 
> > 
> > At this point your phone will likely be freaking out with tons of notifications. I'd disable all sounds before you start, it gets really annoying with it buzzing every 2 seconds.
> > Enable the app again, and install the APK you downloaded. If disable worked, it should install fine.
> > Go to Google Store (before or after install) and disable auto update. I think side loading an APK might prevent auto update anyways.
> > You'll need to re-enable all location services, but just open the Android Auto app and let it do it's thing. This works for me so far, we'll see if it tries to update itself anyways.
> > I got this working on the 7" touchscreen, Raspberry Pi 3 B, standard release SD image, and a Moto Z2 Force phone.
> 
> I roll back the Google Services but after a while they update again. Even after I turned background data usage off

Wont work for me :/ random touches all over the screen and very laggy :/ (Mi6 Xiaomi.eu Rom)

**Comment by peetereczek** [comment]
I just installed the project on my RPI 3+ with custom touch screen.
System works fine with Huawei P30 Pro, as it is supposed to, but also noticed issue with touch only in Android Auto

**Comment by borconi** [comment]
Guys I've posted about this on AASDK, I thought you will pick it up, you need to convert all proto3 to proto2 and recompile the protobuf, that's all.
Sorry I haven't mentioned it here as well... :( 
I've looked it up now and the issue had been removed ( this was the url: https://github.com/f1xpl/aasdk/issues/32 )

**Comment by chucktastic** [comment]
I don't think f1xpl is even remotely interested in backporting fixes to aasdk/openauto from "open"auto pro.  They're too busy charging for it and not releasing any code.

**Comment by borconi** [comment]
I've uploaded the changed protobuf here: https://github.com/borconi/AAProto just change the package name accordingly and recompile should be all working.
It is possible that you will need to adjust the code as well in some places to reflect method changes between proto3 and proto2.

**Comment by chucktastic** [comment]
Wow.  Thank you!  I've been a long time HUR user so I know how much work you've put into supporting the AA community.  I didn't have much luck compiling my own aasdk/openauto a while back.  I may just give it another shot when I've some free time again.

**Comment by roffe** [comment]
why is a protobuf downgrade needed, has someone accidently made a typo and everyone just follows?

Proto3 is the successor of Proto2, why would google downgrade their protobuffers all of a sudden?

**Comment by borconi** [comment]
O don't think it's a downgrade I think it was never intended to be 3 in the first place. The problem is caused by how 2 and 3 are treating empty / 0 fields. Recent Google play services made the checks more strict and where we where able to get away with no value set previously now it needs to see the 0, hence the downgrade needed. Let's not forget that AA was launched officially in 2015 so specs sent to car manufacturers etc are probably dated 2014 or so, that was the time of proto2 not 3.

**Comment by krisstakos** [comment]
> 
> 
> I've uploaded the changed protobuf here: https://github.com/borconi/AAProto just change the package name accordingly and recompile should be all working.
> It is possible that you will need to adjust the code as well in some places to reflect method changes between proto3 and proto2.

Can you gives a bit more detailed instructions how to do it properly?

**Comment by borconi** [comment]
@krisstakos I'm not sure as I'm not really familiar with c/c++ development neither had I looked at all the source code of crankshaft. But generally you need to download the proto compiler from Google, compile the proto to your language (c, java, python, etc) import the created classes in your project and use it.

**Comment by krisstakos** [comment]
I'm a bit lost rigt now. Your proto files are downloaded and stored under aasdk_proto folder. Then I removed proto3 and compiled proto 2.6.1 for the RPI. And when I try to compile aasdk it gives me a ton of errors that can't find proto version and mismatch cimpiler proto3 version. Also I noticed that is some of the files says syntax=proto3 and syntax=proto2 is that normal?

**Comment by borconi** [comment]
> I'm a bit lost rigt now. Your proto files are downloaded and stored under aasdk_proto folder. Then I removed proto3 and compiled proto 2.6.1 for the RPI. And when I try to compile aasdk it gives me a ton of errors that can't find proto version and mismatch cimpiler proto3 version. Also I noticed that is some of the files says syntax=proto3 and syntax=proto2 is that normal?

Never do work at 2AM as the chances are you will do STUPID things. I've uploaded the wrong version of the files, sorry, just pushed a commit now, please re-download and try to compile again you shouldn't see more errors and all files should be proto2 syntax.

**Comment by krisstakos** [comment]
That is the output. Any help to get it going?
protoc --version gives me: libprotoc 2.6.1
Old  libprotobuf 3.0 is removed

sudo cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:357 (file):
  file STRINGS file "/usr/include/google/protobuf/stubs/common.h" cannot be
  read.
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:1 (include)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:363 (math):
  math cannot parse the expression: " / 1000000": syntax error, unexpected
  exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:1 (include)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:364 (math):
  math cannot parse the expression: " / 1000 % 1000": syntax error,
  unexpected exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:1 (include)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:365 (math):
  math cannot parse the expression: " % 1000": syntax error, unexpected
  exp_MOD, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:1 (include)


CMake Warning at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:387 (message):
  Protobuf compiler version 3.0.0 doesn't match library version ..
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:1 (include)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:357 (file):
  file STRINGS file "/usr/include/google/protobuf/stubs/common.h" cannot be
  read.
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:2 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:363 (math):
  math cannot parse the expression: " / 1000000": syntax error, unexpected
  exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:2 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:364 (math):
  math cannot parse the expression: " / 1000 % 1000": syntax error,
  unexpected exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:2 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:365 (math):
  math cannot parse the expression: " % 1000": syntax error, unexpected
  exp_MOD, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:2 (find_package)


CMake Warning at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:387 (message):
  Protobuf compiler version 3.0.0 doesn't match library version ..
Call Stack (most recent call first):
  aasdk_proto/CMakeLists.txt:2 (find_package)


-- Found Protobuf: /usr/lib/arm-linux-gnueabihf/libprotobuf.so;-lpthread;-lpthread (found version "..")
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   system
--   log
--   unit_test_framework
--   date_time
--   log_setup
--   filesystem
--   thread
--   regex
--   chrono
--   atomic
-- Found libusb-1.0:
--  - Includes: /usr/include/libusb-1.0
--  - Libraries: /usr/lib/arm-linux-gnueabihf/libusb-1.0.so
CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:357 (file):
  file STRINGS file "/usr/include/google/protobuf/stubs/common.h" cannot be
  read.
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:363 (math):
  math cannot parse the expression: " / 1000000": syntax error, unexpected
  exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:364 (math):
  math cannot parse the expression: " / 1000 % 1000": syntax error,
  unexpected exp_DIVIDE, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)


CMake Error at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:365 (math):
  math cannot parse the expression: " % 1000": syntax error, unexpected
  exp_MOD, expecting exp_OPENPARENT or exp_NUMBER (2)
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)


CMake Warning at /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:387 (message):
  Protobuf compiler version 3.0.0 doesn't match library version ..
Call Stack (most recent call first):
  CMakeLists.txt:40 (find_package)


-- Found Protobuf: /usr/lib/arm-linux-gnueabihf/libprotobuf.so;-lpthread (found version "..")
-- Configuring incomplete, errors occurred!
See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".

**Comment by borconi** [comment]
@krisstakos - sorry I don't know, I would need to download and understand how the whole crankshaft is built, but I do not have the time and capacity now. 
I think that's the problem here: Protobuf compiler version 3.0.0 doesn't match library version ..
Your protoc isn't quite right, you can use the new protoc compiler to compile proto2 syntax, that's not an issue, as soon as the syntax is changed from 3 to 2 the compile will know what to do.
If you give me the package name I can compile the protobuf to c++ for you, but I need to know the exact package name.

**Comment by tompeel** [comment]
Just as i look to openauto pro life breaths again in the CS community! 
I was taking a look at trying to get this to compile on my PC but im not a dev and am not sure how to open any of it... im used to opening solutions in VS hahah! 
Anyway I commend your work gentlemen!

**Comment by roffe** [comment]
I managed to get `aasdk_proto` to compile on a generic linux host using "proto2" instead of "proto3" in the protocol files. But on a Pi running crankshaft it complains it cannot do it any one. only difference i noted is that the Pi version is running version 3.0.0 of protobuf. sitting and compiling protobuf compiler by hand as we speak and hope for results soon

**Comment by krisstakos** [comment]
> 
> 
> I managed to get `aasdk_proto` to compile on a generic linux host using "proto2" instead of "proto3" in the protocol files. But on a Pi running crankshaft it complains it cannot do it any one. only difference i noted is that the Pi version is running version 3.0.0 of protobuf. sitting and compiling protobuf compiler by hand as we speak and hope for results soon

How you managed to compile successfully? RPI just yelling at me with errors in around 84%

**Comment by borconi** [comment]
Again, I can compile the protos for you to C++ if you give me the exact package name I should use, it will literally take me 2 mins.

**Comment by roffe** [comment]
@krisstakos i copied borconi's proto files into the aasdk_proto folder and ran `cmake CMakeLists.txt` in the repo root, then `make -j3` (-j3 for number of cores on machine -1)

**Comment by roffe** [comment]
no dice

`Jul  4 18:37:21 CRANKSHAFT-NG service_openauto.sh[4279]: /opt/crankshaft/service_openauto.sh: line 182:  4856 Segmentation fault      stdbuf -i0 -o0 -e0 /usr/local/bin/autoapp >> /tmp/openauto.log`

when i connect the phone after rebuilding libaasdk.so & libaasdk_proto.so with @borconi's protobuf files

**Comment by borconi** [comment]
@roffe probably some of the functions needs slight adjustment as well, but I really cannot help there. I know I needed to make some small adjustment in my Java code when I've changed it, but trust me the reason of touch not working is the usage of proto3 instead of proto2.

**Comment by hawkeyexp** [comment]
proto2 build is in progress - stay tuned...
@borconi thx for your hint and help

**Comment by tompeel** [comment]
@hawkeyexp hanging on the end of a thread here! On the plus side i did in this period of broken decided to try Open Auto Pro and have been pretty underwhelmed. Its just not working...

**Comment by borconi** [comment]
@tompeel you should have tried HUR instead 😄  you would been pretty overwhelmed 😁

**Comment by tompeel** [comment]
@borconi whats HUR?

**Comment by borconi** [comment]
HUR = Headunit Reloaded, the app before Crankshaft, basically the same thing but for Android instead of Raspberry. Hence knowing exactly what's the problem as I've faced it in my app as well, about 3 weeks ago, I've just was lucky enough to be on Google Play Service beta so I caught the problem before it became general issue and managed to push out an update for my app before the majority of users got affected. Also that's when I've posted about this on the AASDK warning that this will happen and that it needs addressing, again sorry I haven't warned this community as well.

**Comment by CodeFoodPixels** [comment]
@borconi If they hadn't have hidden the issues on AASDK then it would have been easier for the crankshaft team to see, no blame on you!

**Comment by scylla-xi** [comment]
> proto2 build is in progress - stay tuned...
> @borconi thx for your hint and help

any update?

**Comment by tompeel** [comment]
@borconi im on GPS BETA too, weirdly i had it all working fine with the same GPS version, then i downloaded the new CS on my test pi logged in and it must have flicked a switch because then it wouldn't work with either alpha 4 or 5... i confirmed it with the release date. very odd.

**Comment by abraha2d** [comment]
So before this, I was able to run openauto: the screen/video output worked, but there was no touch input. I merged @borconi's AAProto files into aasdk (great work btw), and made a small change to openauto to get it to compile. Now touch works, but there's no video output (i.e. I can touch the place where the music play button should be, and music starts playing, but I can't actually see anything other than the "Waiting for device" screen).

Anyone have any ideas on what's causing this? These are the changes I made:
https://github.com/f1xpl/aasdk/compare/development...abraha2d:development
https://github.com/f1xpl/openauto/compare/development...abraha2d:development

EDIT: Video output with QtVideoOutput works (laggy and tearing, but it works). It's only OMXVideoOutput that's not working.

EDIT2: Issue solved by using crankshaft-ng, and compiling openauto with fixes there :D

**Comment by sjdean** [comment]
Just as a curiosity, if I take the OpenAuto Pro libaasdk_proto.so and view in a text editor, there's no reference to proto2, only proto3. It makes me wonder if they have fixed the code to call proto3 correctly???

Is this 100% a case that we need to go back to proto2 files, or do we need to fix the code?

**Comment by borconi** [comment]
You cannot really "fix" the code, one of the main diff between 3 and 2 is how 0 are treated. For 3 https://developers.google.com/protocol-buffers/docs/proto3#default it will filter 0 values out (like unset) while proto2 will expect them. Since AA is based on proto2, AFAIK (also proto 3 was only released in 2016, but AA was launched in 2015) we should have never used proto3 in the first place.  

Also I've had trouble with proto3 in my obd2aa project as well where on some cars I've got strange errors and values missing, there I was using protobuffer on the opposite side (on the phone speaking with the car) and data coming from cars was defenetly using proto2 not 3.

**Comment by Richard6360** [comment]
So can we have a build no. 6 with a fix?
For those who canot compile it them selfs?

Op vr 5 jul. 2019 21:51 schreef Emil Borconi-Szedressy <
notifications@github.com>:

**Comment by CodeFoodPixels** [comment]
People need to stop being so rude and impatient. We don't pay the maintainers or contributers of this, and we have no right to demand their time.

**Comment by sjdean** [comment]
Hi Guys, it looks like to get touch working, you only need to swap the tree for InputEventNotificationMessagr from proto3  to proto2. The action_index on TouchEvrnt also needs to be optional. I'm putting together my own fork and will share the changes in one place.

**Comment by borconi** [comment]
@sjdean it's up to you, but I wouldn't mix 3 and 2 and also as AA is based on 2 you never know when and where will be broken because of this... But that's totally your call.

**Comment by sjdean** [comment]
You are totally correct borconi, however for those wishing to take a shortcut and may be faced with other issues or needing to get an older version of protoc, then merely changing this sub set is adequate to get the touch screen working. But yes, the whole lot should be migrated to proto2

**Comment by sli00** [comment]
ok, completely agree that its rude to just ask for a new fixed build a couple days after a fix is discovered, but could someone point me to a tutorial or some information on how to change from proto3 to proto2? I'd love to do it myself I just dont know how

**Comment by gcharlie** [comment]
@sli00 I don't think it's worth anyone's effort to make a tutorial on the matter. Hawkeyexp and others are clearly working to fix the issue, and I'm sure once they have a proper solution in place they will update the code base. Those that are well versed in Android development are putting years of coding experience into fixing this, it's not something that can be learned in a quick how to guide. The rest us can only wait for an update to the app, and maybe add links or suggestions that might help enlighten those doing the heavy lifting. I'm sure this will be fixed, just give it a few days, maybe even week or two or there and all will be right as rain.

**Comment by sli00** [comment]
Alright, thank you, that makes a lot of sense, can't just fix everything with a Youtube video

**Comment by borconi** [comment]
@sli00 It depends how badly you want it. And I'm going to speak from my own experience, in 2016 I had no clue about Java, what's a class, now to compile an APK, etc. I did had some knowledge about PHP and Javascript and HTML but even that is/was limited. That's when I discovered Android Auto, but sadly the Android port I was using which was developer by a great developer, stopped working due to an update in Google Play Service, and the original developer have passed away and nobody was able to fix the issue, so I tried to fix it, and it took me almost 1 month, I was able to come up with able to fix the project, although it did mean I needed to extract the SSL key from a kenwood firmware.  So in one month I've learned how to extract a firmware, how to build and app with NDK, how to binwalk an image, and so on, and all this with StackOverflow and Google, so it is doable but it does need a lot of ambitions and you have to take it step by step and understand what you are doing at each step.

**Comment by abraha2d** [comment]
@sli00 If you're impatient like me, and comfortable compiling openauto from scratch, then I've made the changes necessary on my fork(s): https://github.com/abraha2d/aasdk and https://github.com/abraha2d/openauto. There's build instructions on the f1xpl wiki [here](https://github.com/f1xpl/openauto/wiki/Build-instructions). Replacing /usr/local/bin/autoapp with your compiled version and rebooting should do the trick.

**Comment by sli00** [comment]
Alright I will try that, thank you

**Comment by reganface** [comment]
@abraha2d Compiling from your forks got my setup working again!  Thanks!

**Comment by Ojassy** [comment]
Same here, thanks @abraha2d

**Comment by t00fatt** [comment]
It's not possible to compile this on a rpi2, then boot that sd card with an rpi3B+ is it? I only have a rpi2 available to mess with, and can't take my rpi3B+ in my car out or get my car within wifi range.

**Comment by abraha2d** [comment]
@t00fatt I don't have an RPi3B+ to try it with, but from what I've seen it should work.

**Comment by eduardofilo** [comment]
Here it is a step by step guide for compiling abraha2d forks:

- [In spanish](http://apuntes.eduardofilo.es/2019-07-07-crankshaft-build.html)
- [In english](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9)

Thanks @abraha2d

**Comment by sli00** [comment]
@eduardofilo  @abraha2d  thank you guys so much, touchscreen working now!

**Comment by xxxjajoxxx** [comment]
Cannot write to ‘development.zip’ (Read-only file system).

someone can help with this?

edit:
ok done with this command mount -o remount,rw /

edit2:
new error

cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
-- Could NOT find Protobuf (missing:  Protobuf_INCLUDE_DIR)
CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  aasdk_proto/CMakeLists.txt:2 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".

**Comment by modemlamer** [comment]
> Cannot write to ‘development.zip’ (Read-only file system).
> 
> someone can help with this?
> 
> edit:
> ok done with this command mount -o remount,rw /
> 
> edit2:
> new error
> 
> cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
> -- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
> Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> Call Stack (most recent call first):
> /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
> /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
> aasdk_proto/CMakeLists.txt:2 (find_package)
> 
> -- Configuring incomplete, errors occurred!
> See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
> See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".

you have to unlock the filesystem via

`crankshaft unlock system`

**Comment by xxxjajoxxx** [comment]
> > Cannot write to ‘development.zip’ (Read-only file system).
> > someone can help with this?
> > edit:
> > ok done with this command mount -o remount,rw /
> > edit2:
> > new error
> > cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
> > -- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
> > Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > Call Stack (most recent call first):
> > /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
> > /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
> > aasdk_proto/CMakeLists.txt:2 (find_package)
> > -- Configuring incomplete, errors occurred!
> > See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
> > See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".
> 
> you have to unlock the filesystem via
> 
> `crankshaft unlock system`

still the same error

**Comment by tompeel** [comment]
i just followed the instructions by @eduardofilo  and it worked perfectly. 
I had no FS locking issues, i did remove my card from the pi and edit the config on my pc to avoid messing about with this. 
This is dangerous now i can build! got to get a pi4 up and running

**Comment by eduardofilo** [comment]
> > > Cannot write to ‘development.zip’ (Read-only file system).
> > > someone can help with this?
> > > edit:
> > > ok done with this command mount -o remount,rw /
> > > edit2:
> > > new error
> > > cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
> > > -- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
> > > Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > Call Stack (most recent call first):
> > > /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
> > > /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
> > > aasdk_proto/CMakeLists.txt:2 (find_package)
> > > -- Configuring incomplete, errors occurred!
> > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
> > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".
> > 
> > 
> > you have to unlock the filesystem via
> > `crankshaft unlock system`
> 
> still the same error

You have to put Crankshaft in DEV mode. There are [three possible methods](https://github.com/opencardev/crankshaft/wiki/Dev-Mode-and-Debug-Mode#how-to-enable). I chose to change DEV_MODE param of file `/boot/crankshaft/crankshaft_env.sh` mounting microSD in my computer. The boot partition is FAT32 so you should be able to do even in Windows.

**Comment by eduardofilo** [comment]
> > > Cannot write to ‘development.zip’ (Read-only file system).
> > > someone can help with this?
> > > edit:
> > > ok done with this command mount -o remount,rw /
> > > edit2:
> > > new error
> > > cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
> > > -- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
> > > Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > Call Stack (most recent call first):
> > > /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
> > > /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
> > > aasdk_proto/CMakeLists.txt:2 (find_package)
> > > -- Configuring incomplete, errors occurred!
> > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
> > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".
> > 
> > 
> > you have to unlock the filesystem via
> > `crankshaft unlock system`
> 
> still the same error

It seems as if there is lack of dependences.

Are you working on [last Crankshaft release](https://github.com/opencardev/crankshaft/releases/tag/csng-alpha5)?
Are you installed the packages listed in step 3 of [procedure](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9)? Notice that the lines showed in that step are cropped in the gist view.

**Comment by xxxjajoxxx** [comment]
> > > > Cannot write to ‘development.zip’ (Read-only file system).
> > > > someone can help with this?
> > > > edit:
> > > > ok done with this command mount -o remount,rw /
> > > > edit2:
> > > > new error
> > > > cmake -DCMAKE_BUILD_TYPE=Release ../aasdk
> > > > -- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > > CMake Error at /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
> > > > Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR)
> > > > Call Stack (most recent call first):
> > > > /usr/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
> > > > /usr/share/cmake-3.7/Modules/FindProtobuf.cmake:393 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
> > > > aasdk_proto/CMakeLists.txt:2 (find_package)
> > > > -- Configuring incomplete, errors occurred!
> > > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeOutput.log".
> > > > See also "/home/pi/aasdk_build/CMakeFiles/CMakeError.log".
> > > 
> > > 
> > > you have to unlock the filesystem via
> > > `crankshaft unlock system`
> > 
> > 
> > still the same error
> 
> It seems as if there is lack of dependences.
> 
> Are you working on [last Crankshaft release](https://github.com/opencardev/crankshaft/releases/tag/csng-alpha5)?
> Are you installed the packages listed in step 3 of [procedure](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9)? Notice that the lines showed in that step are cropped in the gist view.

ok i solved previous problems now i got this 

cp: cannot create regular file '/usr/local/bin/autoapp': Text file busy

**Comment by abraha2d** [comment]
@xxxjajoxxx is OpenAuto still running? You may have to quit OpenAuto before you can replace the file. (In dev mode, press the power button on screen, press it again, and run `sudo shutdown -c` in an terminal / SSH connection within 60 seconds to cancel the impending shutdown.)

**Comment by xxxjajoxxx** [comment]
> @xxxjajoxxx is OpenAuto still running? You may have to quit OpenAuto before you can replace the file. (In dev mode, press the power button on screen, press it again, and run `sudo shutdown -c` in an terminal / SSH connection within 60 seconds to cancel the impending shutdown.)

ok done it touch working it was open auto just typed ls -la + kill openauto PID and error disappeared

**Comment by lukeb-uk** [comment]
@hawkeyexp is it worth @abraha2d doing a pull request with his work?

**Comment by abraha2d** [comment]
@lukeb-uk @hawkeyexp I've got PRs open for both [opencardev/aasdk](https://github.com/opencardev/aasdk/pull/1) and [opencardev/openauto](https://github.com/opencardev/openauto/pull/2).

**Comment by CodeFoodPixels** [comment]
Yeah, I realised and deleted my comment. Great work @abraha2d!

**Comment by tompeel** [comment]
This has been forced eye opener for me, I'm interested in getting a bit deeper.
Can anyone advise a place to start looking at this in a Dev environment?
All the Cpp stuff I do is in visual studio, can I open up and compile on there or is it best to build and environment on a pi?

**Comment by JmanNZ** [comment]
Hi
I get the following error after the last make

`In file included from /home/pi/openauto/src/autoapp/UI/MainWindow.cpp:23:0:
/home/pi/openauto_build/ui_mainwindow.h: In member function ‘void Ui_MainWindow::setupUi(QMainWindow*)’:
/home/pi/openauto_build/ui_mainwindow.h:2107:29: error: ‘class QListView’ has no member named ‘setItemAlignment’; did you mean ‘setItemDelegate’?
         AlbumCoverListView->setItemAlignment(Qt::AlignCenter);
                             ^~~~~~~~~~~~~~~~
/home/pi/openauto_build/ui_mainwindow.h:2161:18: error: ‘class QListWidget’ has no member named ‘setItemAlignment’; did you mean ‘setItemWidget’?
         mp3List->setItemAlignment(Qt::AlignCenter);
                  ^~~~~~~~~~~~~~~~
/home/pi/openauto/src/autoapp/UI/MainWindow.cpp: In member function ‘void f1x::openauto::autoapp::ui::MainWindow::on_horizontalSliderVolume_valueChanged(int)’:
/home/pi/openauto/src/autoapp/UI/MainWindow.cpp:767:9: warning: unused variable ‘n’ [-Wunused-variable]
     int n = snprintf(this->volume_str, 5, "%d", value);
         ^
CMakeFiles/autoapp.dir/build.make:734: recipe for target 'CMakeFiles/autoapp.dir/src/autoapp/UI/MainWindow.cpp.o' failed
make[2]: *** [CMakeFiles/autoapp.dir/src/autoapp/UI/MainWindow.cpp.o] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/autoapp.dir/all' failed
make[1]: *** [CMakeFiles/autoapp.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
`

Any help greatly appreciated
Maybe somebody could post the compiled binary

Regards
Jman

**Comment by JoelBrenstrum** [comment]
@JmanNZ 
Here you go

Raspberry Pi 3 Model B  - [autoapp.zip](https://github.com/opencardev/crankshaft/files/3372564/autoapp.zip)

**Comment by JmanNZ** [comment]
@JoelBrenstrum 
Thank you very much

Regards
Jman

**Comment by kooopah** [comment]
After the apparent abandonment of openauto by the developers I have arrived here for the reason of my touchscreen, everything works correctly after recompiling using the proto3 to proto2 reconversion of @borconi (really thanks) System: Ubuntu 18 @ Atomic Pi, protobuf 3.0 , boost 1.65.1, openauto latest version in GIT

**Comment by gcharlie** [comment]
Hey guys, big thanks for the hot fix and all the instruction, but has anybody experienced a dip in audio quality after applying the proto2 fix? This may totally just be my speakers, but I figured I'd ask anyway..

**Comment by GavinAshford** [comment]
> After the apparent abandonment of openauto by the developers I have arrived here for the reason of my touchscreen, everything works correctly after recompiling using the proto3 to proto2 reconversion of @borconi (really thanks) System: Ubuntu 18 @ Atomic Pi, protobuf 3.0 , boost 1.65.1, openauto latest version in GIT

@kooopah did you re-compile openauto after compiling aasdk from @borconi?
If so, which branch and/or who's specifically?
I've taken borconi's aasdk development branch and openauto from @abraha2d and after compiling both openauto app closes whenever the touchscreen it tapped.

N.B. I don't use crankshaft (but it doesn't sound like you do either)

**Comment by SteveGrosbois** [comment]
- I just followed the instructions by @eduardofilo [here](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9) 
- [I also tried the precompiled `autoapp`](https://github.com/opencardev/crankshaft/issues/352#issuecomment-509593494) from @JoelBrenstrum

Same result : it open but it crashes right after i plugged my phone.

My setup is a Raspberry Pi 3+ with official 7" Touch Screen

**Comment by Novocaine85** [comment]
Followed instructions from @eduardofilo and it worked perfectly. Thanks

**Comment by kooopah** [comment]
@GavinAshford Yes, I have always compiled from source adapted to my needs, I do not use crankshaft and I have even built using buildroot through cross compilation, but that's another story ...

use the git of f1xpl for aasdk and openauto, replace the .proto of aasdk proto folder with the broconi version and build it as usual, remember flag of cmake RPI3 = FALSE and everything should go well

**Comment by GavinAshford** [comment]
@kooopah awesome, thanks I'll give that a go this evening! Thinking about it now my issue might be the flag - the Pi in my car is Pi3 but the one I'm using to compile on is 2B+ because I don't really want to dismantle my car to get the 3 out...

**Comment by krisstakos** [comment]
> 
> 
> Followed instructions from @eduardofilo and it worked perfectly. Thanks

How did you compile the openauto? It always gives me this:
`In file included from /home/pi/openauto/src/autoapp/UI/MainWindow.cpp:23:0:
/home/pi/openauto_build/ui_mainwindow.h: In member function ‘void Ui_MainWindow::setupUi(QMainWindow*)’:
/home/pi/openauto_build/ui_mainwindow.h:2109:29: error: ‘class QListView’ has no member named ‘setItemAlignment’; did you mean ‘setItemDelegate’?
         AlbumCoverListView->setItemAlignment(Qt::AlignCenter);
                             ^~~~~~~~~~~~~~~~
/home/pi/openauto_build/ui_mainwindow.h:2163:18: error: ‘class QListWidget’ has no member named ‘setItemAlignment’; did you mean ‘setItemWidget’?
         mp3List->setItemAlignment(Qt::AlignCenter);
                  ^~~~~~~~~~~~~~~~
/home/pi/openauto/src/autoapp/UI/MainWindow.cpp: In member function ‘void f1x::openauto::autoapp::ui::MainWindow::on_horizontalSliderVolume_valueChanged(int)’:
/home/pi/openauto/src/autoapp/UI/MainWindow.cpp:767:9: warning: unused variable ‘n’ [-Wunused-variable]
     int n = snprintf(this->volume_str, 5, "%d", value);
         ^
CMakeFiles/autoapp.dir/build.make:734: recipe for target 'CMakeFiles/autoapp.dir/src/autoapp/UI/MainWindow.cpp.o' failed
make[2]: *** [CMakeFiles/autoapp.dir/src/autoapp/UI/MainWindow.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/autoapp.dir/all' failed
make[1]: *** [CMakeFiles/autoapp.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
`

**Comment by Novocaine85** [comment]
@krisstakos i Just follwed the English instructions provided by @eduardofilo and everything went well. 

I don't know if it does any difference but do you have a pi3 or pi2?

**Comment by krisstakos** [comment]
I followed exactly the instructions. I have pi3

**Comment by krisstakos** [comment]
> 
> 
> @krisstakos i Just follwed the English instructions provided by @eduardofilo and everything went well.
> 
> I don't know if it does any difference but do you have a pi3 or pi2?

I followed exactly the instructions. I have pi3  
btw: aasdk is compiling fine

edit: I commented out mentioned lines and it compiles successfully

**Comment by abraha2d** [comment]
@GavinAshford borconi's aasdk repo is missing a few fixes that cause crashes. You can see the fixes missing here: https://github.com/borconi/aasdk/compare/development...abraha2d:development.

~~I've opened another PR to borconi's repo, but in the meantime you can use mine.~~
EDIT: @borconi's repo has the changes now.

**Comment by abraha2d** [comment]
Even if you only have a Pi2, you can leave the RPI3 flag as true, otherwise you loose the hardware acceleration. Things will still work, but very slowly.

**Comment by abraha2d** [comment]
@krisstakos make sure you have Qt v5.12 or greater. Looks like `setItemAlignment` was added in 5.12, so if your Qt version is older than that, the compile will fail.

**Comment by borconi** [comment]
@abraha2d I've committed the pull request

**Comment by GavinAshford** [comment]
> Even if you only have a Pi2, you can leave the RPI3 flag as true, otherwise you loose the hardware acceleration. Things will still work, but very slowly.

@abraha2d I ended up using your dev branches for both aasdk and openauto yesterday - it now no-longer crashes when touching (I might have used your crankshaft-ng branch the day before, I can't remember exactly).  But I compiled without the RPI3 flag and as you say, it no-longer has hardware acceleration.
I'll re-compile this evening with the flag on and hopefully it'll be back to normal.

**Comment by SteveGrosbois** [comment]
@GavinAshford In the instructions by @eduardofilo [here](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9), it uses the `crankshaft-ng` branch for openauto.

But, for me, after I plug my phone, Android Auto opens less than one second and crashes immediatly.

Which branch do you use ? `crankshaft-ng` is the the most up to date branch.

**Comment by borconi** [comment]
@kwiky sudden crash can be caused by one of the protos data marked as required but not being supplied, for example GPS data you don't set the timestamp, that will cause a crash on Java so potentially on c++ as well.

**Comment by SteveGrosbois** [comment]
@borconi How to debug this (to eventually fix it) ?

**Comment by JmanNZ** [comment]
> I just followed the instructions by @eduardofilo here
> I also tried the precompiled autoapp from @JoelBrenstrum
> 
> Same result : it open but it crashes right after i plugged my phone.
> My setup is a Raspberry Pi 3+ with official 7" Touch Screen

I had the same issue with the autoapp from @JoelBrenstrum
I managed to compile it now and it works so thanks heaps to everyone for the awesome work
My working autoapp hopefully this helps somebody else
[autoapp.zip](https://github.com/opencardev/crankshaft/files/3383578/autoapp.zip)

Regards
Jman

**Comment by borconi** [comment]
@kwiky I'm not sure about C++ as I've never worked with it, in Java (using Android Studio) it's pretty simple because you will the crash in the logs instantly pointing to the line of code which caused it, then you just change the according proto and recompile the proto.

**Comment by gregorhollmig** [comment]
Is crankshaft going to be continued? 
Is there already another community fork where source code can be merged and new builds created?
@htruong could you merge please?

**Comment by CodeFoodPixels** [comment]
That's a shitty way to react to a project that people do in their spare time.

**Comment by abraha2d** [comment]
@kwiky check `/tmp/openauto.log`, the error should be printed in there somewhere (along the lines of "Can't parse message of type "WhateverTypeTheMessageIs" because it is missing required fields: required_field_1, required_field_2", etc, if indeed it is a protobuf error)

**Comment by gregorhollmig** [comment]
> That's a shitty way to react to a project that people do in their spare time.

@lukeb-uk Sorry I didn't mean to offend anyone. Code has been written which is awesome! but as far as I can see the community cannot use it in this project as of right now? (the owner was not actively contributing on GitHub since April)
https://github.com/opencardev/openauto/pull/2 https://github.com/opencardev/aasdk/pull/1

I like the project and use it daily, I would like to contribute here more, but what is the way forward for contribution? There is no discussion on both PRs but also no merge.

**Comment by gregorhollmig** [comment]
> That's a shitty way to react to a project that people do in their spare time.

Sorry that was going in the wrong direction.
I would love to see the communities effort merged into the projects around crankshaft. (https://github.com/opencardev/aasdk/pull/1 https://github.com/opencardev/openauto/pull/2)

**Comment by GavinAshford** [comment]
> @GavinAshford In the instructions by @eduardofilo [here](https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9), it uses the `crankshaft-ng` branch for openauto.
> 
> But, for me, after I plug my phone, Android Auto opens less than one second and crashes immediatly.
> 
> Which branch do you use ? `crankshaft-ng` is the the most up to date branch.

@kwiky I'm 90% sure I cloned the -development branches from @abraha2d for both aasdk and openauto
However it looks like the development branch for openauto has gone now... @abraha2d did you have a development branch of openauto a couple of days ago, or am I mis-remembering??

**Comment by abraha2d** [comment]
@GavinAshford you're not mis-remembering, I messed around with my repos a little bit. They're now forked from opencardev/aasdk and opencardev/openauto.
The old `development` branch was fixes applied to @f1xpl's original openauto.
What you want is the `crankshaft-ng` branch for openauto, which is fixes applied to the latest openauto as present in Crankshaft. On the aasdk repo, `development` branch is still the latest (there's nothing crankshaft specific in there).

**Comment by GavinAshford** [comment]
Ah that explains it! Your old development branch (which I cloned a couple of days ago, before you removed it) works for me - I don't use crankshaft, just raw openauto. I guess in that situation the f1xpl original + some fixes is what is needed.
I think when I tried the crankshaft-ng branch that's the one that seemed to run OK but crashed on any screen press.

**Comment by abraha2d** [comment]
@GavinAshford I'll push it back up, I think I still have it locally.
EDIT: It's back up now.

**Comment by ThomasCorrea** [comment]
Is there any solution?? How is it compiled? I have been searching and trying to learn for days and I can not find

**Comment by adamoutler** [comment]
I can see there is a solution which requires manual compiling if you want the fix.

I'm beginning to think that the project has become derelict.  It has been over a month since release with a critical show-stopping issue affecting all users.  I think it might be time to find another solution.

**Comment by CodeFoodPixels** [comment]
@adamoutler How about being constructive and asking if there's any way you can help instead?

**Comment by adamoutler** [comment]
> @adamoutler How about being constructive and asking if there's any way you can help instead?

I was being constructive.  The person above me had a question about the solution.  I told him there was a solution.  I answered 1/2 of his concern.   Here's the other 1/2  https://github.com/borconi/aasdk/compare/development...abraha2d:development  

Additionally, I added my personal thoughts.  I think this project is not keeping up with Google.  I reverted my car stereo system to stock for this, but I feel running crankshaft is unmaintainable and the crankshaft project appears unmaintained.  Even with a solution available, there are no images for over a month.  There is no official fix pushed out to images or updates. 

I mean, anyone who is installing this for the first time needs to know this.  The last time OpenAuto was maintained was in May.  Since then it was updated to a new version of Debian, but no issues have been addressed in regards to touchscreen.

**Comment by JiseOT** [comment]
Crankshaft is open source project so everyone can add fixes and improvements. Unfortunately I am definitely not a tech guy and after first problem with Google services couple months ago I decided to migrate to pro version of openauto just to prevent ending up with broken head unit someday. Now I see that was worth it because Pro version receives updates very often, there is a direct contact with developers that implement user suggestions and no risk that you will end up with broken head unit for a month or forever. Anyway, I still wish all the best for this project!

**Comment by adamoutler** [comment]
I think that's what I am looking for. https://bluewavestudio.io/index.php/bluewave-shop/openauto-pro-detail

**Comment by gcharlie** [comment]
I'm pretty surprised at how long its taken to get the touch screen issue addressed on the main release, especially since  @abraha2d and friends have come up with a solution that provides a pretty good fix. seems like it would be a matter of just merging that work in. Or at the very least some communication as to why that hasn't happened already would help keep the community in the loop.
@adamoutler, with that said I don't think it's fair to compare a paid product to a community project. We all know Open Auto Pro is more stable and feature complete; that is why they charge for it.

**Comment by ThomasCorrea** [comment]
Who solved the problem, can you share the files? already resolved

**Comment by gcharlie** [comment]
@ThomasCorrea, @abraha2d did in their fork, but it hasn't been pulled into the core branches, so you have to compile it yourself. It's a somewhat involved process but @eduardofilo posted a very in-depth line by line instructions on how to go about doing that https://github.com/opencardev/crankshaft/issues/352#issuecomment-509029490. it took me about 30 minutes or so, but all you need is to be able to put your crankshaft pi into dev mode, and another computer to SSH into it from.

**Comment by Stargazzer** [comment]
Hello everyone, I think this project is great. Yes an update/ETA on a new version would be helpful.  I am an extreme noob when it comes to Linux command line.  I can follow directions well but if I don't know the terminology then I can't complete it.  I am stuck on Step # 4 "Download abraha2d's aasdk fork and compile:" If there are steps on how to do that I am sure I could use the solution provided.

I can wait for a new version of Crankshaft to be released.
I would also love to try to fix it with the solution provided here if I had an explanation on how to do Step 4 which would also allow me to do Step 6.
Thanks.

**Comment by chucktastic** [comment]
A couple of individuals have compiled the file and offered it up for download.  Boot crankshaft into dev mode, make a copy of /usr/local/bin/autoapp and copy the extracted autoapp file to /usr/local/bin/.  Make sure the ownership and permissions match the other files in the folder.  Reboot back to normal mode and see if it works again.


https://github.com/opencardev/crankshaft/issues/352#issuecomment-509593494
https://github.com/opencardev/crankshaft/issues/352#issuecomment-510621673

**Comment by Stargazzer** [comment]
> A couple of individuals have compiled the file and offered it up for download. Boot crankshaft into dev mode, make a copy of /usr/local/bin/autoapp and copy the extracted autoapp file to /usr/local/bin/. Make sure the ownership and permissions match the other files in the folder. Reboot back to normal mode and see if it works again.
> 
> [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-509593494)
> [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-510621673)

Thanks for being Chucktastic. :) I will give this a try.

**Comment by ThomasCorrea** [comment]
as you will see I am new in this .. How do I start crankshaft in dev mode? 
@gcharlie Thanks you!

**Comment by eduardofilo** [comment]
> as you will see I am new in this .. How do I start crankshaft in dev mode?
> @gcharlie Thanks you!

In [this document](https://github.com/opencardev/crankshaft/wiki/Dev-Mode-and-Debug-Mode) there are three methods, although I think that number 2 does not work. I used the first one, editing the file `/boot/crankshaft/crankshaft_env.sh` mounting the card in my computer.

**Comment by ThomasCorrea** [comment]
I followed the steps of @eduardofilo and in the installation of the packages I get this error

3. Install some packages:

$ sudo apt-get install -y libboost-all-dev libusb-1.0.0-dev libssl-dev cmake libprotobuf-dev protobuf-c-compiler protobuf-compiler

E: Could not open lock file / var / lib / dpg / lock - open (13: permission denied)
E: Unable to locate the administration directory (/ var / lib / dpgk), are you rooted?

**Comment by aftjah** [comment]
> > as you will see I am new in this .. How do I start crankshaft in dev mode?
> > @gcharlie Thanks you!
> 
> In [this document](https://github.com/opencardev/crankshaft/wiki/Dev-Mode-and-Debug-Mode) there are three methods, although I think that number 2 does not work. I used the first one, editing the file `/boot/crankshaft/crankshaft_env.sh` mounting the card in my computer.

Method #2 appears to work only if the stick is formatted as FAT32.

**Comment by aftjah** [comment]
> A couple of individuals have compiled the file and offered it up for download. Boot crankshaft into dev mode, make a copy of /usr/local/bin/autoapp and copy the extracted autoapp file to /usr/local/bin/. Make sure the ownership and permissions match the other files in the folder. Reboot back to normal mode and see if it works again.
> 
> [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-509593494)
> [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-510621673)

I had no luck with the zipped autoapp on my Pi 3 B (continuous reboots), but I rebuilt autoapp according to  @eduardofilo's instructions and everything works great.

**Comment by Aaron407s** [comment]
I followed the compiling instructions listed by @eduardofilo to the letter and it worked great! Big thanks for taking the time to lay everything out for those of us who are not as savvy on the programming side!

**Comment by modemlamer** [comment]
> > A couple of individuals have compiled the file and offered it up for download. Boot crankshaft into dev mode, make a copy of /usr/local/bin/autoapp and copy the extracted autoapp file to /usr/local/bin/. Make sure the ownership and permissions match the other files in the folder. Reboot back to normal mode and see if it works again.
> > [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-509593494)
> > [#352 (comment)](https://github.com/opencardev/crankshaft/issues/352#issuecomment-510621673)
> 
> I had no luck with the zipped autoapp on my Pi 3 B (continuous reboots), but I rebuilt autoapp according to @eduardofilo's instructions and everything works great.

i did the same. works great.
just follow this guide, step by step.
https://gist.github.com/eduardofilo/eced05b5fcafb86178834f4d708dcfa9

i put my zipped image(1.2G) to my cloud storage. contact me to get the download link

**Comment by crstmkt** [comment]
Hey @modemlamer 
just tried to contact you but somehow this didn't work. Would like to get the zipped image from you. Could you please provide the download link?

**Comment by modemlamer** [comment]
--> felix@funits.de

**Comment by alextytan** [comment]
Cheers, mate!

**Comment by miro279** [comment]
I have the greatest respect for the devs who develop open source projects and crankshaft in particular. Still this bug has been around for 5 weeks now and makes the software unusable, also a fix is around for those who are willing to dig deeper..
I can completlly understand if live does not allow to put loads of time in projects for free!
Anyways I think a some point people deserve to know if this project has a future or not, not meaning this disrespectful at all.

**Comment by Maju3** [comment]
@miro279 I  agree withyou.
Sure that normal life of the developer has priority but a simple statement about the future of crankshaft would be fair.
I mean just 1 minute to write here should be possible or?

**Comment by modemlamer** [comment]
I'm sure he said that already in another thread. This project isn't dead. Just wait and be patient. There are guides and files to get it running. And yes, u have to digg into it. 
Turnkey solutions often are not 4 free. 
There is also a stepByStep Guide u have just to follow. If this guide is to complex for u, OK, just wait or search for another project.

**Comment by Maju3** [comment]
Don"t get me wrong but I think the lack of information about crankshafts future development is one if the reason why many people switch to the paid openauto pro version.
And if I remember right was the general idea of Huan Trong and Hawkeyexp a software project what needs no tinkering and linux knowledge.

**Comment by gregorhollmig** [comment]
@Maju3 @miro279 problem is, this is a currently a very small group of developers that cannot be too active. @htruong said himself in one of the open pull requests, he cannot handle it. It would be great to open this project to a more broad dev community with the ability to contribute. 
Some maintainers are required and some kind of roadmap what would be important goals (e.g. automatic builds for RPI systems, Theming, extension of sensor support, better documentation and website ... so many things).

**Comment by Maju3** [comment]
Sure.
It is just that Hawkeyexp is too quiet here in my opinion.
He is the main developer and his latest comment about this issue is 26 days old.
Even there is allready a solution for this issue.
So I wonder why he do not include this solution into his build and release a new crankshaft version?

**Comment by Maju3** [comment]
I really hate to say this but it seems to me that if you want quick bug fixes you need to have linux skills or you need to pay for openauto pro.

**Comment by modemlamer** [comment]
Come'on it's summertime, he's maybe on Holiday, or maybe moving elsewhere...
Beeing offline a month is not that thing..

**Comment by modemlamer** [comment]
Android and raspbian are based in Linux, so yes u have to have decent Linux skills... Otherwise you better pay for a caraudio that has this Feature built in...

**Comment by Maju3** [comment]
Or maybe he lost interrest in this and thats it.
Time will tell.

**Comment by Maju3** [comment]
@modemlamer
You are right that android and raspian are based on linux but that does not mean that I need linux skills to use it.
Just think about libreelec or kodi.
Everyone who is able to burn an image to a sd card can use it, right ;)
And exactly this point was the reason why I like crankshaft.
Just at the moment it is useless without digging deeper.

**Comment by gcharlie** [comment]
@Maju3 is bring up some good points. As someone who's been following this project in mostly a passive manner, I know not to expect devs to get every bug squashed immidiately, thats not realistic and it's not fair to them.
But with that said, when a month goes by with not a single word on a bug thats breaking functionality thats core to the project thats not great.
It's just a matter of managing expectations.
I think if we got an update along the lines of "Hey guys I know this bug is breaking things bad, but I'm on vacation and we're not going to be able to get a solid fix in for at least 2 months" or "...hey guys sorry but  I think it makes more sense in the long run to tackle this after the next big Android Auto update (because thats probably going to break things even harder)".
I'd be perfectly happy with that and I think most other folks would be too.

**Comment by Maju3** [comment]
@gcharlie
Absolutely correct!
I do not expect that a bug can be fixed in one or two days.
But I think it would be only fair to let people know what the current state of crankshaft is.
Like you say something like " guys i know that this bug makes crankshaft nearly useless but these days i am busy with other things....stay tuned..."

**Comment by abraha2d** [comment]
From what I've heard @htruong and @hawkeyexp are both somewhat busy with life. I've signed on to help maintain this project. Still waiting on write access to the repos, but hopefully we'll have a official build out soon.

**Comment by htruong** [comment]
@abraha2d You now have access to the repos :) Thank you for offering help!

**Comment by abraha2d** [comment]
Ok everyone, thanks for being patient. OTA updates are now available with the latest fixes!

If your Raspberry Pi running Crankshaft-NG is connected to the internet, you should see "Update available!" in the top-right corner of the screen. Press it, and then press "Update OpenAuto" to get the fixes.

(Also, fyi Crankshaft-NG still works with the new Android Auto interface, which is good!)

**Comment by Stargazzer** [comment]
Kevin,

That is great news, now will this be the only way to update in the future?
Or will there still be the file to download to flash to sdcard option?

Thanks,
Stargazzer77

**Comment by eduardofilo** [comment]
> Kevin, That is great news, now will this be the only way to update in the future? Or will there still be the file to download to flash to sdcard option? Thanks, Stargazzer77

Kevis is also preparing a release:

https://github.com/opencardev/crankshaft/releases/tag/csng-alpha5.1

**Comment by abraha2d** [comment]
@Stargazzer The plan is to release Alpha5.1 with the updates baked in. It's out here: https://github.com/opencardev/crankshaft/releases/tag/csng-alpha5.1

I haven't had the time to flash it to an sdcard and test it yet, though. In theory it should work, but it's my first time dealing with the crankshaft build system, and I'm not sure whether I've messed anything up.

**Comment by Bishamon1987** [comment]
I'll test it today. I damaged something with a test to run to instances of
camoverlay in x11. Crankshaft doesn't run in x11 anymore since camoverlay
crashed... Very weird.


Op vr 9 aug. 2019 18:34 schreef Kevin Abraham <notifications@github.com>:

**Comment by Mandrew1444** [comment]
I will test, give me around an hour

**Comment by Mandrew1444** [comment]
Can confirm, the update fixed the touchscreen in open auto for me. Essential PH1

**Comment by abraha2d** [comment]
Personally tested Alpha 5.1, and it seems to be working fine (at least USB mode, hardware incompatibility prevents me from testing WiFi mode).

For those of you having issues with openauto crashing in X11 mode, see #366. Specifically [this comment](https://github.com/opencardev/crankshaft/issues/366#issuecomment-520072487).

**Comment by Ayybeeshafi** [comment]
> I got this working for the time, so I figured I'd share.
> 
> First you'll need to download the google play services APK version for your phone, here's the guide I used (pay attention to the sub version numbers):
> https://itechify.com/2018/01/12/install-right-google-play-services-android-device/
> And the apk downloads link:
> https://google-play-services.en.uptodown.com/android/versions
> 
> I tried `17.7.81` but that didn't work, I ended up using `17.5.26`.
> Here's the settings I changed:
> 
> * `Settings > Google Services > Location` turn off location entirely
> * `Settings > Google Services > Security > Find my device` turn off entirely
> * `Settings > Apps > App Info > Google Play Services` first disable, then force stop
> * `Settings > Apps > App Info > Google Play Services > Storage > Manage Space`  Clear all data, then on the previous screen Clear Cache
> 
> At this point your phone will likely be freaking out with tons of notifications. I'd disable all sounds before you start, it gets really annoying with it buzzing every 2 seconds.
> 
> Enable the app again, and install the APK you downloaded. If disable worked, it should install fine.
> 
> Go to Google Store (before or after install) and disable auto update. I think side loading an APK might prevent auto update anyways.
> 
> You'll need to re-enable all location services, but just open the Android Auto app and let it do it's thing. This works for me so far, we'll see if it tries to update itself anyways.
> 
> I got this working on the 7" touchscreen, Raspberry Pi 3 B, standard release SD image, and a Moto Z2 Force phone.

This seemed to work for me.
